### ISSUE TITLE
The "force locale" option is not working

### ISSUE THREAD

**Issue by asereze** [AnySoftKeyboard/AnySoftKeyboard#1489]
### Steps to reproduce
1.Opened the app
2. Opened the "Tweaks" option menu and the "Force language" option
3. Selected a language different from my system's one

### Expected behaviour
To get the app's strings in the chosen language

### Actual behaviour
The locale doesn't change. I still get my system's one.

Android OS version:
Android 8.1 (lineage OS 15.1)
App version: 1.9.2445

Device manufacturer and model:
Xiaomi Redmi Note 4X

List of installed add-ons (like languages, or themes):
No themes, four languages (Italian, English, Portugues and the Sardinian one I created)

**Comment by Mikaela** [comment]
I see this on OnePlus 3 with Android 8.0 (OP3_O2_BETA_39) and Nokia 1 with Android Go 8.1.0 on Nokia 1 (TA-1047).

I was going to request this feature, but instead searched for issues and found this, because the tweaks button is a bit hidden.

1.9.2481from Play Store Beta

**Comment by ArenaL5** [comment]
Already fixed, please close.

**Comment by asereze** [comment]
@ArenaL5 No, In the last version available (1.10.606) the problem is still there.

**Comment by ArenaL5** [comment]
Oh, I think the see the problem. The "(takes a while to take effect)" note is misleading.

You need to refresh the UI. Meaning: Force locale doesn't translate text currently on screen, but ensures that any future configuration or menu you open will be translated.

To see the text on your chosen language, you can either open a different configuration screen from the button bar (but that will leave the button bar itself untranslated) or click the □ Overview key on Android, so you can close AnySoftKeyboard and open it again.

I've tried this on version 1.10.606 from F-Droid, and the debug version. Please tell us if it's working or if you're having further problems.

**Comment by asereze** [comment]
@ArenaL5 I tried doing that, but even then nothing changed. Even if I completely close the app or restart my device everything remains in my system's language. I tried that with different languages, and to do a test I've set up the app in Portuguese a week ago but I still haven't seen a string in that language (my system's one is Italian).

**Comment by ArenaL5** [comment]
Thanks for telling the languages you're using, we're closer to solving this. My testing phone doesn't have an Italian locale, but I tried selecting a random language and then selecting European Portuguese and Brazilian Portuguese in AnySoftKeyboard. 

It seems like translations to both European and Brazilian Portuguese are broken. Brazilian Portuguese includes English texts, and European Portuguese translates very little text. Any text string that ASK can't find in your current forced locale won't be translated, and will default either to the last text displayed (in whatever language), or to your system locale.

The Gestures screen should have _some_ text in Portuguese changes header. The missing translations either default to your system locale, or to the last forced locale.

I couldn't see any problem until now because I'm switching between English and Spanish; Spanish is mostly complete, and English is the first language to be programmed right now. Until now I only tried with Spanish and English. Trying either one of these languages should translate everything, or almost everything.

Can you confirm everything I said? Does the app behave differently in your phone?

If this is the problem, the solution is to submit the correct translations to https://crowdin.com/project/anysoftkeyboard and then wait for the developers to include them in for the next version.

**Comment by asereze** [comment]
@ArenaL5 I've tried both English and Spanish, and neither one worked. Everything is still in Italian, unfortunately.

**Comment by ArenaL5** [comment]
This is weird... I can't replicate the bug. Perhaps it has something to do with LineageOS; my phones are running on stock Android.

Which version of AnySoftKeyboard are you running, exactly? You said the version number is 1.10.606, but where did you download it from? (Google Play, F-Droid, GitHub, compiled it yourself from GitHub?) Does it come from a regular, alpha or beta channel?

**Comment by asereze** [comment]
@ArenaL5 I downloaded it from Google Play, it's the regular channel.

**Comment by GoranGharib** [comment]
You right the App Language is not changes by that option *force locale* that needs by fix i had this problem many time ago when i have a Android 7.1.1 Device!!

**Comment by ArenaL5** [comment]
@asereze I tried uninstalling my version from F-Droid and installing it from Play Store. I still can't reproduce it as you described it; there's always some text translated for me.

@GoranGharib Are you using stock Android or LineageOS? Whichever you're using, could you test it now?

**Comment by ArenaL5** [comment]
**Update:** I couldn't exactly reproduce your bug, but I found something else. When turning your phone, strings return to their original locale.
